### DISCUSSION TITLE
Distinguishing between default values and user-provided values

### DISCUSSION THREAD

**Discussion by multimeric** [fastapi/typer#668] [Show and tell]
I had a scenario where it was important to work out whether a given value was provided by the user on the command line, or if it was a default, because user-provided values needed to have higher precedence in argument compilation. This took me a lot of searching, but there's actually a very simple way to do this with typer:

```python
from typer import Context
from click.core import ParameterSource

@app.command
def main(
    arg_a: str = "foo",
    arg_b: int = 123,
    # ... etc
    context: Context
):
    if context.get_parameter_source("arg_a") == ParameterSource.COMMANDLINE:
        # Do something here
```

The other `ParameterSource` options are listed here: https://click.palletsprojects.com/en/8.1.x/api/#click.core.ParameterSource

**Comment by svlandeg** [comment]
Thanks for sharing! 🙏
